velocity/src/main/java/me/leoko/advancedban/velocity/VelocityMethods.java

```diff
@@ -0,0 +1,377 @@
+      log("[AdvancedBan] Offline permission support through LuckPerms active");
+    } else {
+      luckPermsSupport = false;
+      log("[AdvancedBan] No offline permission support through LuckPerms");
+    }
+
+  }
+
+
+  @Override
+  protected ConfigurationNode loadConfiguration(Path configPath) throws IOException {
+      YAMLConfigurationLoader loader = YAMLConfigurationLoader.builder().setPath(configPath).build();
+      return loader.load();
+  }
+
+  @Override
+  public String getFromUrlJson(String url, String key) {
+    try {
+      HttpURLConnection request = (HttpURLConnection) new URL(url).openConnection();
+      request.connect();
+
+      JsonObject json = (JsonObject) JsonParser.parseReader(new InputStreamReader(request.getInputStream()));
+
+      String[] keys = key.split("\\|");
+      for (int i = 0; i < keys.length - 1; i++) {
+        json = json.getAsJsonObject(keys[i]);
+      }
+
+      return json.get(keys[keys.length - 1]).toString().replaceAll("\"", "");
+
+    } catch (Exception exc) {
+      return null;
+    }
+  }
+
+  @Override
+  public String getVersion() {
+    return "@version";
+  }
+
+  // Object.toString() is considered safe in this scenario (as per Configurate Docs)
+  @Override
+  public String[] getKeys(Object file, String path) {
+    return ((ConfigurationNode)file).getNode(path.split("\\.")).getChildrenMap().keySet().stream().map(Object::toString).toArray(String[]::new);
+  }
+
+  @Override
+  public void setupMetrics() {
+    // no bstats for velocity yet
+  }
+
+
+  @Override @Deprecated
+  public boolean isBungee() {
+    return false;
+  }
+
+  @Override
+  public boolean isProxy() {
+    return true;
+  }
+
+  @Override
+  public ServerType getServerType() {
+    return ServerType.VELOCITY;
+  }
+
+  @Override
+  public String clearFormatting(String text) {
+    return text.replaceAll("(&[^\\s])+", "");
+  }
+
+  @Override
+  public Object getPlugin() {
+    return VelocityMain.get();
+  }
+
+  @Override
+  public File getDataFolder() {
+    return dataDirectory.toFile();
+  }
+
+  @Override
+  public void setCommandExecutor(String cmd, TabCompleter tabCompleter) {
+    CommandMeta meta = server.getCommandManager().metaBuilder(cmd).build();
+    server.getCommandManager().register(meta, new CommandReceiverVelocity(server, cmd));
+  }
+
+  @Override
+  public void sendMessage(Object player, String msg) {
+      ((CommandSource) player).sendMessage(LegacyComponentSerializer.legacyAmpersand().deserialize(msg));;
+  }
+
+  @Override
+  public String getName(Object player) {
+    if (player instanceof Player) {
+      return ((Player) player).getUsername();
+    } else {
+      return "CONSOLE";
+    }
+  }
+
+  @Override
+  public String getName(String uuid) {
+      return server.getPlayer(UUID.fromString(uuid)).get().getUsername();
+  }
+
+  @Override
+  public String getIP(Object player) {
+    return ((Player)player).getRemoteAddress().getAddress().getHostAddress();
```

Comment:
```suggestion
    return ((Player)player).getRemoteAddress().getHostName();
```
For conformity with BukkitMethods and BungeeMethods